internal/connector/internal/handlers/processor_agent.go

```diff
@@ -0,0 +1,215 @@
+package handlers
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/connector/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/connector/internal/api/private"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/connector/internal/presenters"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/handlers"
+	"github.com/getsentry/sentry-go"
+	"github.com/golang/glog"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services"
+	"github.com/gorilla/mux"
+)
+
+func (h *ConnectorClusterHandler) ListProcessorDeployments(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	query := r.URL.Query()
+	connectorClusterId := mux.Vars(r)["connector_cluster_id"]
+
+	cfg := &handlers.HandlerConfig{
+		Validate: []handlers.Validate{
+			handlers.Validation("connector_cluster_id", &connectorClusterId, handlers.MinLen(1), handlers.MaxLen(maxConnectorClusterIdLength), validateConnectorClusterId(ctx, h.Service)),
+		},
+		Action: func() (interface{}, *errors.ServiceError) {
+
+			gtVersion := int64(0)
+			if v := query.Get("gt_version"); v != "" {
+				gtVersion, _ = strconv.ParseInt(v, 10, 0)
+			}
+
+			listArgs := services.NewListArguments(query)
+
+			getList := func() (list private.ProcessorDeploymentList, err *errors.ServiceError) {
+
+				resources, paging, err := h.Service.ListProcessorDeployments(ctx, connectorClusterId, false, false, false, listArgs, gtVersion)
+				if err != nil {
+					return
+				}
+
+				list = private.ProcessorDeploymentList{
+					Kind:  "ProcessorDeploymentList",
+					Page:  int32(paging.Page),
+					Size:  int32(paging.Size),
+					Total: int32(paging.Total),
+				}
+
+				for _, resource := range resources {
+					converted, serviceError := h.resolveProcessorDeploymentRefsAndPresentDeployment(&resource)
+					if serviceError != nil {
+						sentry.CaptureException(serviceError)
+						glog.Errorf("Failed to present processor deployment %s: %v", resource.ID, serviceError)
+						// also reduce size and total count
+						list.Size--
+						list.Total--
+					} else {
+						list.Items = append(list.Items, *converted)
+					}
+				}
+				return
+			}
+
+			if v := query.Get("watch"); v == "true" {
+				idx := 0
+				list, err := getList()
+				bookmarkSent := false
+
+				sub := h.Bus.Subscribe(fmt.Sprintf("/kafka_connector_clusters/%s/processor_deployments", connectorClusterId))
```

Comment:
To be honest I'm not entirely clear of the life-cycle in RHOC.

I have the `processor_mgr.go` reconciliation of `assigning` create a `ProcessorDeployment` which is then picked up by this (event)bus subscription. `ProcessorDeployment`'s then become visible to the FleetShard through the FleetManager agent calls. I'll have to write some tests to confirm the operation.

An explanation of the status/phase transition mechanism would be helpful.